build-utils/mount-partitions.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/sh
+
+# Mounts boot, state, & data partitions from balenaOS.
+# The container must be privileged for this to function correctly.
+
+use_legacy_message="Using legacy host mounts."
+
+# Get the current boot device in case there are duplicate partition labels
+# for `(balena|resin)-(boot|state|data)` found. Because Supervisor is started with
+# `--net=host`, /etc/resolv.conf is bind-mounted by the Engine into the container,
+# so the partition where resolv.conf is located is assumed to be one of the
+# partitions of the current active device. The partition identifer `pN` is removed 
+# from the end so we can compare with the base device.
+current_boot_device=$(grep /etc/resolv.conf /proc/mounts | awk '{print substr($1,1,length($1)-2)}')
+if [ "${current_boot_device}" = "" ]; then
+    echo "WARNING: Could not determine boot device. Please launch Supervisor as a privileged container with host networking."
```

Comment:
This should probably exit with an error, there might be no point to fallback to using /mnt/root if the container is not provileged

Reply:
Added `exit 1` in [latest force-push](https://github.com/balena-os/balena-supervisor/pull/2120/commits/71cddb6a1b9d51a59a350726a80035e5afce1403#diff-8a96bc9af661488f86b92785e9382315f61c548974c4bcce92c1a1deb9a129f8R13-R14).